pkg/controller/podautoscaler/replica_calculator_test.go

```diff
@@ -972,8 +1066,30 @@ func TestReplicaCalcToleranceCMObject(t *testing.T) {
 
 func TestReplicaCalcTolerancePerPodCMObject(t *testing.T) {
 	tc := replicaCalcTestCase{
-		currentReplicas:  4,
-		expectedReplicas: 4,
+		currentSpecReplicas:   4,
+		currentStatusReplicas: 4,
+		expectedReplicas:      4,
+		metric: &metricInfo{
+			metricType:              objectPerPodMetric,
+			name:                    "qps",
+			levels:                  []int64{20166},
+			perPodTargetUtilization: 5000,
+			expectedUtilization:     5042,
+			singleObject: &autoscalingv2.CrossVersionObjectReference{
+				Kind:       "Deployment",
+				APIVersion: "apps/v1",
+				Name:       "some-deployment",
+			},
+		},
+	}
+	tc.runTest(t)
+}
+
+func TestReplicaCalcTolerancePerPodCMObjectSurge(t *testing.T) {
+	tc := replicaCalcTestCase{
+		currentSpecReplicas:   4,
+		currentStatusReplicas: 8,
```

Comment:
What would happen if we had 0 in either spec or status (or both)?

Reply:
These test case fields are misleading.  We don't do anything with `statusReplicas` except write it down any more.  So a zero value is uninteresting.  Likewise, `specReplicas` is either left unmodified (in which case no big deal) or set to the desired scale.  In this case, zero already has the meaning, don't scale.  So nothing will happen anyway.  We already have a unit tests for this: https://github.com/kubernetes/kubernetes/blob/9120d427dab4a69ddcf22597f81b574fb1af96bc/pkg/controller/podautoscaler/horizontal_test.go#L1633

I've renamed these as `specReplicas` and `actualPodCount`.  The actual pod count is just used by the test to setup mocks.